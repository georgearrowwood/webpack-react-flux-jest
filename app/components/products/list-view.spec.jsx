import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';

import ProductsList from './list-view';


const mockProductsList = [
  { id: 1, title: 'test product 1' },
  { id: 2, title: 'test product 2' },
];

test('Render products list', () => {
  const wrapper = mount(<ProductsList
    products={mockProductsList}
    deleteProductHandler={() => {}}
  />);
  console.log('sn', wrapper);
  
  expect(wrapper).toMatchSnapshot();
  // expect(wrapper.prop('products')).arrayContaining(mockProductsList);
  // console.log('p', wrapper.prop('products'));
  
  // expect(wrapper.prop('product')).toMatchSnapshot();
});
