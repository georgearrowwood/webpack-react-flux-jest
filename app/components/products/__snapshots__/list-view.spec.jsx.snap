// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render products list 1`] = `
<Products
  deleteProductHandler={[Function]}
  products={
    Array [
      Object {
        "id": 1,
        "title": "test product 1",
      },
      Object {
        "id": 2,
        "title": "test product 2",
      },
    ]
  }
>
  <div>
    List:
    <ul>
      <li
        key="1"
      >
        1
         - 
        test product 1
         :
        <a
          href="/product/delete"
          onClick={[Function]}
        >
          X
        </a>
      </li>
      <li
        key="2"
      >
        2
         - 
        test product 2
         :
        <a
          href="/product/delete"
          onClick={[Function]}
        >
          X
        </a>
      </li>
    </ul>
  </div>
</Products>
`;
